### PR TITLE
Ensuring UTF-8 encoding for MSVC builds

### PR DESCRIPTION
**Pull request by shellygr**
Support for different locales in WIN compilation by setting /utf-8 flag in MSVC.

Build may otherwise fail with errors such as:
`solidity\test\libdevcore\utf8.cpp : warning C4819: The file contains a character that cannot be represented in the current code page (1255). Save the file in Unicode format to prevent data loss`

### CODE REVIEW

File: CompilerOptions.cmake [hunk 1]
```diff
@@ -0,0 +1,4 @@
+if (MSVC)
+    # add /utf-8 flag to compiler options
+    SET(CMAKE_CXX_FLAGS_INIT "${CMAKE_CXX_FLAGS_INIT} /utf-8")
```
Comment (axic) [inline CompilerOptions.cmake]: There is a `CompilerOptions` file already in `cmake/EthCompilerSettings.cmake`. Can you move this there please?

Comment (shellygr) [reply CompilerOptions.cmake]: Thanks for the reference - I pushed an update.

**Comment by axic** [comment]
@shellygr thanks, can you squash the commits into one?

What MSVC versions did required this flag?

**Comment by shellygr** [comment]
Since it's a one line change I'll close this pull request and open another one. 
Re. your question - it's VS2017. Not sure this option existed before that.

**Comment by axic** [comment]
If you create PRs from branches it is usually easier to squash/rebase/.. work with them.

**Pull request closed without merging**